torch/csrc/api/include/torch/optim/adagrad.h

```diff
@@ -18,40 +18,79 @@ class InputArchive;
 namespace torch {
 namespace optim {
 
-struct TORCH_API AdagradOptions {
+struct TORCH_API AdagradOptions : public OptimizerCloneableOptions<AdagradOptions> {
   AdagradOptions(double learning_rate);
   TORCH_ARG(double, learning_rate);
   TORCH_ARG(double, lr_decay) = 0;
   TORCH_ARG(double, weight_decay) = 0;
+  TORCH_ARG(double, initial_accumulator_value) = 0;
+  TORCH_ARG(double, eps) = 1e-10;
+public:
+  ~AdagradOptions() = default;
```

Comment:
I am actually not 100% sure. If we don't define the destructor for the subclasses then when the pointers to the base class are deleted, only the base class is called. What I am worried about it if it can cause an undefined behavior in some cases. Thoughts?

Reply:
that's a good point, let's keep it to be safe :D